File: FrontEnd/src/services/patientService.js
import api from "./api";

export const getAllPatients = async () => {
    const response = await api.get("/patients");
    return response.data;
};

export const getPatientById = async (id) => {
    const response = await api.get(`/patients/${id}`);
    return response.data;
};

export const createPatient = async (patient) => {
    const response = await api.post("/patients", patient);
    return response.data;
};

export const updatePatient = async (id, patient) => {
    const response = await api.put(`/patients/${id}`, patient);
    return response.data;
};

export const deletePatient = async (id) => {
    await api.delete(`/patients/${id}`);
};

File: PatientService/src/main/java/com/infosys/MediSphereApp/service/PatientService.java
package com.infosys.MediSphereApp.service;

import com.infosys.MediSphereApp.model.Patient;
import java.util.List;

public interface PatientService {

    Patient savePatient(Patient patient);

    List<Patient> getAllPatients();

    Patient getPatientById(String id);

    Patient updatePatient(String id, Patient patient);

    void deletePatient(String id);
}